### ISSUE TITLE
Don't include pictures in the repository

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#5122]
I would like to discuss the strategy for including pictures and videos in the HoloViz docs and in Panel in particular.

## Motivation

I just git pull'ed a branch from Panel and I can see that a lot of the pictures (and gifs?) I've included in some PRs are now a part of the repository

![image](https://github.com/holoviz/panel/assets/42288570/f3fb2aa3-8201-43d7-bb8e-b0f2d4866ff3)

I thought they would be moved to a CDN? As far as I can see some of the `.gif`s have also been converted to `.png`s.

## Questions

1. Should assets like images, data and video be included in the repository?
  -  pro: easy to do, easy to version
  - cons: makes the repository heavy to git clone
2. Do we want to include videos in the docs.
  -  pro: we can illustrate interactivity (like templates, full apps)
  - cons: the docs become less useful as interactive notebooks. They take time to create and maintain.
3. If we include videos should we do this as .gif or .mp4?
  - pro: the advantage of .gif is that they autoplay
  - cons: the disadvantage of .gif is that they are (much) larger than .mp4

**Comment by MarcSkovMadsen** [comment]
Regarding 1. I don't like including assets in the repository. I really like having a repository that is fast to git clone. I've had repositories that took minutes to git clone and its not a nice user experience and can also increases the time of your ci/cd.

**Comment by philippjfr** [comment]
I think we should include images in the repo where it makes sense and host most gifs/videos on the CDN. This is what I did in your streamlit PR, the total size of all the images I added was <50 KB. Assets in the repo are easier to keep in sync and as long as they're small won't have a significant impact.

I don't have strong feelings about gifs vs videos but so far we've always used gifs.

I did not convert any of your gifs to images.

**Comment by droumis** [comment]
I dislike the use of the CDN for assets. It's a pain for fast-changing docs. I guess it makes sense for a small number of large videos/gifs that don't change often.

In general, I think the very selective use of videos/gifs can convey a concept that's particularly complicated to show in a static image or interactive plot (like a sequence of click-through events), but I don't think they should become the default way we convey information.

**Comment by MarcSkovMadsen** [comment]
We need a `ScreenRecorder` component @maximlt . Like Streamlit has.

- You can add it to a layout.
- It can be activated via a button
- You can configure it to produce either .mp4 or .gif
- You can configure it to rescale the image to some width x height format
  - Optionally you could make it change the window size to that width x height format
- You can configure it to compress the video
- The video should be automatically downloaded or made available for download

**Comment by maximlt** [comment]
I wa referring to Desktop/CLI/web app tools and practices.

**Comment by MarcSkovMadsen** [comment]
Yeah. But we still need a `ScreenRecorder` and `AudioRecorder` if we want to enable users with machinelearning and Generative AI use cases.

**Comment by maximlt** [comment]
Yep. But that's a separate request ;)

**Comment by ahuang11** [comment]
`cons: the disadvantage of .gif is that they are (much) larger than .mp4`

I think they can be comparable if you use a tool like [gifsicle](https://github.com/kohler/gifsicle) to optimize the GIF.

**Comment by philippjfr** [comment]
Indeed, I optimize all gifs we upload with gifsicle.
